### ISSUE TITLE
update bot to apply label "state: community" to PRs submitted by people outside Angular team

### ISSUE THREAD

**Issue by jenniferfell** [angular/angular#24829]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[X ] Other... Please describe: Angular automation and tooling
</code></pre>

## Current behavior

There is no easy way to search for all PRs contributed by community members/volunteers. 

## Expected behavior

Be able to easily find all PRs contributed by community members/volunteers.

## What is the motivation / use case for changing the behavior?

Goal: Be able to easily find all open docs PRs submitted by people who are not part of the Angular team (aka who are community contributors). Be able to use this feature to more efficiently support, process, and run reports on community contributions.

Currently, the docs team has been doing this manually, using a label called "state: community WIP".
- If someone from the community has volunteered to work on an issue (or is already working on it and has submitted a PR), the docs team manually applies the state: community WIP label to that issue.
- When we identify a docs PR submitted by someone from the community, we apply the state: community WIP label to that PR.

We determine whether or not to apply the state: community WIP label based on our knowledge of who is on the Angular team. 

We'd like to automate some of what we've been doing manually.

We'd like a bot to do what we've been doing manually:

1. Go through all open PRs with label “comp: docs” 
https://github.com/angular/angular/pulls?q=is%3Aopen+is%3Apr+label%3A%22comp%3A+docs%22

2. Check the author of each to see if that person is part of the Angular team., perhaps by checking to see if the author belongs to the Angular org or group? 

3. If the author is not part of Angular team, then add label “state: Community WIP”.

4. (bonus step) If the PR is linked to an associated issue, then add state: community WIP label to that issue also. 

Wa la! 

We need this soon, to help with the issue and PR backlog. If it takes too long, we will have completed the work manually.

## Environment

<pre><code>
Angular version: X.Y.Z

Browser:
- [ ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by JoostK** [comment]
Why include WIP? That may not be accurate for pull requests by community members

**Comment by trotyl** [comment]
Agree with @JoostK, a community PR is most likely just waiting for review rather than in progress.

**Comment by jenniferfell** [comment]
We initially created the label for issues, so we could begin to have a way to search/filter on which issues have associated work being done by community members, so we don't duplicate effort and we can offer support if needed. The label already exists and is in use. 

I see your point. I'll ponder a rename now that we are using it more generally to include PRs as well, in all levels of completeness. I may not find perfect wording, but I hope the intent is reasonably clear for now. 

Thanks for noticing. :-)

**Comment by jenniferfell** [comment]
@JoostK and @trotyl : Re the label name. How about if we change “state: community WIP” to just “state: community” or "state: community work"

Other things I considered: 

- "comp:" is used for the area of Angular being worked on, not the team doing the work
- “Flag: “ prefix feels like a level of alert or a problem to be corrected. I don't want to create alarm. 
- “State: Community WIP” worked perfectly for issues (where I started using it), but as you point out WIP isn’t as appropriate for PRs that generally begin not as WIP but in the review state. 
- “State: community submission” doesn’t work for issues because most are submitted by the community. We wanted one label to handle both issues and PRs. 

I realize this isn't a big deal, but I just thought I'd see if you have a preference. :-)

**Comment by jenniferfell** [comment]
@trotyl Thanks. I pondered some combinations like the one you suggested, and we had some more internal discussion. We plan to use hotlist: community-help (or something similar) as a way to highlight issues for which we'd most greatly appreciate community help, but that means that label is on the issue (not the PR). It also means that there will still be PRs submitted by contributors that aren't explicitly solicited via "hotlist: community-help" label. 

So...for now, I changed it to state: community. 

Thanks again
